spec/utils/db_producer_spec.rb

```diff
@@ -155,6 +155,10 @@
         Deimos.configure { |c| c.db_producer.compact_topics = [] }
       end
 
+      it 'should not raise NoMethodError if all the keys are unique' do
+        Deimos.configure { |c| c.db_producer.compact_topics = %w(my-topic my-topic2) }
+        expect { producer.compact_messages(deduped_batch) }.not_to raise_error(NoMethodError)
+      end
```

Comment:
I'd say this is kind of a silly way to name and focus the test - the point is not that it does not raise this specific error - you fixed the thing that would cause this. You just need to make sure that this code runs in this situation, so you could say something like "it should compact messages where all messages are unique" and check the result. Automatically it will also check that it doesn't raise an error, because if it does, the spec will fail. :)

Reply:
Ah yeah you are right, as always! I will update the spec :)